definitions/checks/candlepin/validate_candlepin_db.rb

```diff
@@ -0,0 +1,20 @@
+module Checks::Candlepin
+  class ValidateCandlepinDb < ForemanMaintain::Check
+    metadata do
+      description 'Check to validate candlepin database'
+      tags :pre_upgrade
+      label :validate_candlepin_db
+
+      confine do
+        feature(:candlepin) &&
+          feature(:candlepin_database).check_option_using_cpdb_help('validate') != 0
+      end
+    end
+
+    def run
+      feature(:candlepin_database).execute_cpdb_validate_cmd
+      output = execute('echo $?').strip.to_i
```

Comment:
As you are using `execute!` in `execute_cpdb_validate_cmd`, isn't it by itself already raising an exception on non-zero exit code?

Reply:
Yes, right.
`execute! ` already raising an exception on non-zero exit code.